parcels/interaction/base_neighbor.py

```diff
@@ -0,0 +1,126 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+from parcels.interaction.distance_utils import spherical_distance
+
+
+class BaseNeighborSearch(ABC):
+    """Base class for searching particles in the neighborhood."""
+    name = "unknown"
+
+    def __init__(self, interaction_distance, interaction_depth,
+                 max_depth=100000):
+        self.interaction_depth = interaction_depth
+        self.interaction_distance = interaction_distance
+        self.inter_dist = np.array(
+            [self.interaction_distance, self.interaction_distance,
+             self.interaction_depth]).reshape(3, 1)
+        self.max_depth = max_depth
+        self._values = None
+        self._active_mask = None
```

Comment:
@qubixes : could you please explain a bit what this `_active_mask` is storing or representing ? I already saw that this mask is used quite often, but it would be good to know what it masks. Does the mask only `True`-include particles that moved between time-steps, or does it mask-out particles that are not-yet-started, or what is the semantic meaning of this variable ?

Reply:
Same as the other comment on this, let me know where you want to have more documentation/comments.